File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/RequestTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class RequestTyped {
    public String method;
    public String page;
    public String address;
    public String benchmark;
    public String format;

    public RequestTyped(String method, String page, String address, String benchmark, String format) {
        this.method = method;
        this.page = page;
        this.address = address;
        this.benchmark = benchmark;
        this.format = format;
    }

    public RequestTyped(RequestString s) {
        this.method = s.method;
        this.page = s.page;
        this.address = s.address;
        this.benchmark = s.benchmark;
        this.format = s.format;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof RequestTyped)) return false;
        RequestTyped that = (RequestTyped) o;
        return Objects.equals(method, that.method)
            && Objects.equals(page, that.page)
            && Objects.equals(address, that.address)
            && Objects.equals(benchmark, that.benchmark)
            && Objects.equals(format, that.format);
    }

    @Override
    public int hashCode() {
        return Objects.hash(method, page, address, benchmark, format);
    }

    public RequestString toRequestString() {
        return new RequestString(String.valueOf(method), String.valueOf(page), String.valueOf(address), String.valueOf(benchmark), String.valueOf(format));
    }

    public static List<RequestTyped> fromStringList(List<RequestString> list) {
        List<RequestTyped> result = new ArrayList<>();
        for (RequestString s : list) result.add(new RequestTyped(s));
        return result;
    }

    public static List<RequestString> toStringList(List<RequestTyped> list) {
        List<RequestString> result = new ArrayList<>();
        for (RequestTyped t : list) result.add(t.toRequestString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("method", method);
        m.put("page", page);
        m.put("address", address);
        m.put("benchmark", benchmark);
        m.put("format", format);
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static RequestTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new RequestTyped(
            Json.asString(Json.require(m, "method"), "method"),
            Json.asString(Json.require(m, "page"), "page"),
            Json.asString(Json.require(m, "address"), "address"),
            Json.asString(Json.require(m, "benchmark"), "benchmark"),
            Json.asString(Json.require(m, "format"), "format"));
    }

    public static RequestTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<RequestTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (RequestTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<RequestTyped> fromJSONList(String json) {
        List<RequestTyped> result = new ArrayList<RequestTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "RequestTyped")));
        return result;
    }
}

File: CSharp/SpecStudioExamplesTestsCSharp/common/RequestTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class RequestTyped
    {
        public string method;
        public string page;
        public string address;
        public string benchmark;
        public string format;

        public RequestTyped(string method, string page, string address, string benchmark, string format)
        {
            this.method = method;
            this.page = page;
            this.address = address;
            this.benchmark = benchmark;
            this.format = format;
        }

        public RequestString ToRequestString()
        {
            return new RequestString(
                Json.ToText(this.method),
                Json.ToText(this.page),
                Json.ToText(this.address),
                Json.ToText(this.benchmark),
                Json.ToText(this.format)
            );
        }

        public static List<RequestString> ToStringList(List<RequestTyped> list)
        {
            var result = new List<RequestString>();
            foreach (var t in list) result.Add(t.ToRequestString());
            return result;
        }

        public static List<RequestTyped> FromStringList(List<RequestString> list)
        {
            var result = new List<RequestTyped>();
            foreach (var s in list) result.Add(s.ToRequestTyped());
            return result;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WriteString("method", this.method);
            w.WriteString("page", this.page);
            w.WriteString("address", this.address);
            w.WriteString("benchmark", this.benchmark);
            w.WriteString("format", this.format);
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static RequestTyped FromJsonElement(JsonElement m)
        {
            return new RequestTyped(
                Json.AsString(Json.Require(m, "method"), "method"),
                Json.AsString(Json.Require(m, "page"), "page"),
                Json.AsString(Json.Require(m, "address"), "address"),
                Json.AsString(Json.Require(m, "benchmark"), "benchmark"),
                Json.AsString(Json.Require(m, "format"), "format")
            );
        }

        public static RequestTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<RequestTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<RequestTyped> FromJSONList(string json)
        {
            var result = new List<RequestTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "RequestTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"Method={method}, Page={page}, Address={address}, Benchmark={benchmark}, Format={format}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not RequestTyped other) return false;
            return object.Equals(this.method, other.method)
                && object.Equals(this.page, other.page)
                && object.Equals(this.address, other.address)
                && object.Equals(this.benchmark, other.benchmark)
                && object.Equals(this.format, other.format);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.method);
            h.Add(this.page);
            h.Add(this.address);
            h.Add(this.benchmark);
            h.Add(this.format);
            return h.ToHashCode();
        }
    }
}
